src/otx/backend/native/engine.py

```diff
@@ -153,6 +151,11 @@ def __init__(
         self._model: OTXModel = (
             model if isinstance(model, OTXModel) else self._auto_configurator.get_model(**get_model_args)
         )
+        if self.checkpoint:
+            if not isinstance(self.checkpoint, (Path, str)) and not Path(self.checkpoint).exists():
+                msg = f"Checkpoint {self.checkpoint} does not exist."
+                raise FileNotFoundError(msg)
+            self.model.load_state_dict_incrementally(torch.load(self.checkpoint))
```

Comment:
Could you move the checkpoint logic closer to the top, where the other variables are initialised? I think something like this would be nice:
```py
if checkpoint:
    if not isinstance(checkpoint, (str, Path)) and not Path(checkpoint).exists():
        raise FileNotFoundError(f"Checkpoint file not found: {checkpoint_path}")
    self.checkpoint = checkpoint
    self.model.load_state_dict_incrementally(torch.load(checkpoint))
```

Reply:
I moved the assignment of the checkpoint closer to where it is loaded into the model. However, it’s not possible to move it further up, since the model depends on the AutoConfigurator and DataModule being instantiated first. The model requires information from the DataModule, so those components must be initialized before the model is created.